projects/igniteui-angular/src/lib/time-picker/time-picker.component.ts

```diff
@@ -1792,7 +1795,7 @@ export class IgxTimePickerComponent implements
      * ```
      */
     public cancelButtonClick(): void {
-        if (this.mode === InteractionMode.DropDown) {
+        if (this.mode === InteractionMode.DropDown && this.value) {
```

Comment:
Yeah, this doesn't quite do it - if there's no initial value(null, empty, etc), opening the drop-down will auto-populate the display with the currently selected time and this won't allow it to be cleared on cancel and still trigger invalid event for 4AM?